tests/js/widgets.js

```diff
@@ -0,0 +1,106 @@
+    expect(container.props().children[1].type).toBe(Column);
+    expect(container.props().children[2].props.children).toBe('c');
+    expect(container.props().children[2].type).toBe(Column);
+    expect(columns.text()).toBe('<Container />');
+});
+
+
+test('Column render', () => {
+    let column = shallow(<Column />);
+    expect(column.text()).toBe('');
+    let div = column.find('div');
+    expect(div.length).toBe(1);
+    expect(div.props().className).toBe('column');
+    column = shallow(<Column>xyz</Column>);
+    expect(column.text()).toBe('xyz');
+    column = shallow(<Column className='special-column'></Column>);
+    div = column.find('div');
+    expect(div.length).toBe(1);
+    expect(div.props().className).toBe('column special-column');
+});
+
+
+test('Container render', () => {
+    let container = shallow(<Container />);
+    expect(container.text()).toBe('');
+    expect(container.instance().columnStyles).toEqual([]);
+    let div = container.find('div');
+    expect(div.props().className).toBe('container');
+    expect(div.props().style).toEqual(
+        {"content": "", "display": "table", "width": "100%"});
+    expect(div.props().children).toEqual([]);
+    container = shallow(<Container columns={3} />);
+    expect(container.text()).toBe('');
+    div = container.find('div');
+    expect(container.instance().columnStyles).toEqual(
+        [{"boxSizing": "border-box", "float": "left", "overflow": "hidden", "width": "33.333333333333336%"},
+         {"boxSizing": "border-box", "float": "left", "overflow": "hidden", "width": "33.333333333333336%"},
+         {"boxSizing": "border-box", "float": "left", "overflow": "hidden", "width": "33.333333333333336%"}]);
+    expect(div.props().children).toEqual([]);
+    expect(div.props().style).toEqual(
+        {"content": "", "display": "table", "width": "100%"});
+    expect(div.props().children).toEqual([]);
+    container = shallow(<Container columns={3} ratios={[2, 3, 7]} />);
+    expect(container.text()).toBe('');
+    expect(container.instance().columnStyles).toEqual(
+        [{"boxSizing": "border-box", "float": "left", "overflow": "hidden", "width": "16.666666666666664%"},
+         {"boxSizing": "border-box", "float": "left", "overflow": "hidden", "width": "25%"},
+         {"boxSizing": "border-box", "float": "left", "overflow": "hidden", "width": "58.333333333333336%"}]);
+});
+
+
+test('Container render children', () => {
+    const children = [<div>FOO...</div>, <div>...BAR</div>]
+    const styles = jest.fn(() => {
+        return [
+            {background: 'black', color: 'red'},
+            {background: 'green', color: 'gold'}]})
+
+    class MockContainer extends Container {
+
+        get columnStyles () {
+            return styles();
+        }
+    }
+
+    const container = shallow(<MockContainer children={children} />);
+    expect(container.text()).toBe('FOO......BAR');
+    expect(styles.mock.calls).toEqual([[], [], [], []])
```

Comment:
I know that We talked about the guidelines in code that if linters don't complain then it's okay.
But I find a little bit challenging to understand what should be and de facto is tested.
I feel that We could improve situation here by doing one of 2 things.:
1. Create more smaller test-cases with bigger and more verbose descriptions.
2. Add newlines and add comments per every section of given test ->

Reply:
im just switching on eslint for new js code if that helps re linting

- smaller test cases - in some cases ok - but its expensive to build/teardown tests whether its db/rollback or react rendering - so i try and get stuff done while they are built/rendered

- new lines - i can go for a bit of new lineage - but not so much - its a pita try and read more code at the same time